src/zenml/integrations/azure/orchestrators/azureml_orchestrator.py

```diff
@@ -0,0 +1,455 @@
+        )
+
+    def get_orchestrator_run_id(self) -> str:
+        """Returns the run id of the active orchestrator run.
+
+        Important: This needs to be a unique ID and return the same value for
+        all steps of a pipeline run.
+
+        Returns:
+            The orchestrator run id.
+
+        Raises:
+            RuntimeError: If the run id cannot be read from the environment.
+        """
+        try:
+            return os.environ[ENV_ZENML_AZUREML_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_AZUREML_RUN_ID}."
+            )
+
+    @staticmethod
+    def _create_command_component(
+        step: Step,
+        step_name: str,
+        env_name: str,
+        image: str,
+        command: List[str],
+        arguments: List[str],
+    ) -> CommandComponent:
+        """Creates a CommandComponent to run on AzureML Pipelines.
+
+        Args:
+            step: The step definition in ZenML.
+            step_name: The name of the step.
+            env_name: The name of the environment.
+            image: The image to use in the environment
+            command: The command to execute the entrypoint with.
+            arguments: The arguments to pass into the command.
+
+        Returns:
+            the generated AzureML CommandComponent.
+        """
+        env = Environment(name=env_name, image=image)
+
+        outputs = {"completed": Output(type="uri_file")}
+
+        inputs = {}
+        if step.spec.upstream_steps:
+            inputs = {
+                f"{upstream_step}": Input(type="uri_file")
+                for upstream_step in step.spec.upstream_steps
+            }
+
+        return CommandComponent(
+            name=step_name,
+            display_name=step_name,
+            description=f"AzureML CommandComponent for {step_name}.",
+            inputs=inputs,
+            outputs=outputs,
+            environment=env,
+            command=" ".join(command + arguments),
+        )
+
+    def _create_or_get_compute(
+        self, client: MLClient, settings: AzureMLOrchestratorSettings
+    ) -> Optional[str]:
+        """Creates or fetches the compute target if defined in the settings.
+
+        Args:
+            client: the AzureML client.
+            settings: the settings for the orchestrator.
+
+        Returns:
+            None, if the orchestrator is using serverless compute or
+            str, the name of the compute target (instance or cluster).
+        """
+        # If the mode is serverless, we can not fetch anything anyhow
+        if settings.mode == AzureMLComputeTypes.SERVERLESS:
+            return None
+
+        # If a name is not provided, generate one based on the orchestrator id
+        compute_name = settings.compute_name or f"compute_{self.id}"
+
+        # Try to fetch the compute target
+        try:
+            client.compute.get(compute_name)
```

Comment:
What if the fetching here gets a compute instance but the orchestrator is configured to run with a compute cluster? Or the attributes don't match the ones defined in the orchestrator? I remember us talking about it, but don't remember if this was something that is not possible with the current AzureML SDK

Reply:
Alright, I made a bunch of changes here and I am quickly going to test them now, but while I do that, let me sum it up here:

- Added a validator to the settings class, this way if you set a value that the mode of operation does not support it throws a warning.
- Checking if the mode of the settings matches the compute target if we are working with existing compute instances and compute clusters. In the case they don't match, we are throwing a `RuntimeError` now.
- In the case where you are working with a compute target that already exists, there is a chance that you still have some additional configuration (left from the time you created the compute target). We are only throwing a warning in this case, if the settings do not match the actual configuration of the compute target. 
- Finally, I am checking whether a compute cluster is provisioned successfully/a compute instance is in a running state, before executing the pipeline. We can probably add some logic here to restart them if that is not the case, but I would do it in a separate PR as I am not so sure about all the possible cases here.

Reply:
Tested most of these use cases already and they are working as intended.